### PR TITLE
Added Prepare action for subnet

### PR DESCRIPTION
**Pull request by Am018**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by number213** [comment]
CI failure is caused by incorrect specs, this PR fixes it: https://github.com/Azure/azure-rest-api-specs/pull/5013

**Review by jianghaolu — APPROVED** [review]

**Comment by jianghaolu** [comment]
Will wait for 5013 to be merged to re run ci

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualNetwork.json [hunk 1]
```diff
@@ -1376,6 +1431,50 @@
...
+          },
+          "description": "A list of NetworkIntentPolicyConfiguration."
+        }
+      }
+    },
+    "NetworkIntentPolicyConfiguration": {
+      "properties": {
+        "networkIntentPolicyName": {
+          "type": "string",
+          "description": "The name of the Network Intent Policy for storing in target subscripton."
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualNetwork.json]: subscripton -> subscription

Comment (Am018) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualNetwork.json]: Thanks for catching that. I have taken care of it in latest update

**Comment by Am018** [comment]
Can this PR be merged now since all the checks are passing and I have taken care of all the comments?

**Comment by Am018** [comment]
Thanks @jianghaolu

**Pull request merged**